scripts/retrieve_block/retrieve_block.go

```diff
@@ -0,0 +1,279 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/internal/pubip"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/common/variadic"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/host"
+	lip2pnetwork "github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+type chainSpec struct {
+	Bootnodes  []string `json:"bootnodes"`
+	ProtocolID string   `json:"protocolId"`
+}
+
+func setupP2PClient() host.Host {
+	const port = 30333
+
+	// create multiaddress (without p2p identity)
+	listenAddress := fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", port)
+	addr, err := ma.NewMultiaddr(listenAddress)
+	if err != nil {
+		log.Fatalf("creating multiaddress: %s", err.Error())
+	}
+
+	var externalAddr ma.Multiaddr
+	ip, err := pubip.Get()
+	if err != nil {
+		log.Fatalf("getting public ip: %s", err.Error())
+	}
+
+	externalAddr, err = ma.NewMultiaddr(fmt.Sprintf("/ip4/%s/tcp/%d", ip, port))
+	if err != nil {
+		log.Fatalf("creating external address: %s", err.Error())
+	}
+
+	// Set your own keypair
+	priv, _, err := crypto.GenerateKeyPair(
+		crypto.Ed25519, // Select your key type. Ed25519 are nice short
```

Comment:
This is a manual step needed? Should this be highlighted in the readme?

Reply:
Oh, removed the comments, is not needed to do any manual step